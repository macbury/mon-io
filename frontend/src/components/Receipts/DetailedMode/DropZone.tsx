import React, { useRef, useCallback } from 'react'
import { View } from 'react-native'
import { Text, TouchableRipple } from 'react-native-paper'
import styled from 'styled-components/native'

const DropZoneContainer = styled(TouchableRipple)`
  display: flex;
  flex-direction: column;
  opacity: 0.6;
  border: 4px;
  border-radius: 10px;
  border-style: dashed;
  cursor: pointer;
  border-color: ${({ theme }) => theme.headerBorderColor};
  margin: 20px 20px 0px 20px;
`

const DropZoneText = styled(Text)`
  text-align: center;
  justify-content: center;
  font-size: 20px;
  font-weight: bold;
  min-height: 120px;
  align-items: center;
  display: flex;
`

export interface IDropZoneProps {
  children: any;
  onFileSelect(file : Array<File>)
}

export default function DropZone({ children, onFileSelect } : IDropZoneProps) {
  const fileInputRef = useRef<HTMLInputElement>()

  const selectFile = useCallback(() => {
    if (fileInputRef.current) {
      fileInputRef.current.click()
    }
  }, [fileInputRef?.current])

  const onFileChange = useCallback((event : React.ChangeEvent<HTMLInputElement>) => {
    event.stopPropagation();
    event.preventDefault();
    const file = event.target.files[0];
    onFileSelect([file]);
  }, [onFileSelect])

  return (
    <DropZoneContainer onPress={selectFile}>
      <View>
        <input
          type="file"
          ref={fileInputRef}
          style={{display: 'none'}}
          onChange={onFileChange}
        />

        <DropZoneText>{children}</DropZoneText>
      </View>
    </DropZoneContainer>
  )
}