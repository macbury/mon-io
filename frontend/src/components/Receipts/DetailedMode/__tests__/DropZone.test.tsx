import React from 'react'
import { shallow } from 'enzyme'
import snapshot from 'enzyme-to-json'

import DropZone from '../DropZone'

it('renders correctly', () => {
  const component = shallow(<DropZone onFileSelect={jest.fn()}>Hello world</DropZone>)
  expect(snapshot(component)).toMatchSnapshot()
})