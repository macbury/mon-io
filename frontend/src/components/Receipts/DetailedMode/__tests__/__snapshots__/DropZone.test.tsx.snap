// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(withTheme(TouchableRipple))
  onPress={[Function]}
>
  <View>
    <input
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <Styled(withTheme(Text))>
      Hello world
    </Styled(withTheme(Text))>
  </View>
</Styled(withTheme(TouchableRipple))>
`;
